WordPress/WordPressTest/PostRepositoryPostsListTests.swift

```diff
@@ -0,0 +1,191 @@
+import Foundation
+import XCTest
+import OHHTTPStubs
+
+@testable import WordPress
+
+class PostRepositoryPostsListTests: CoreDataTestCase {
+
+    var repository: PostRepository!
+    var blogID: TaggedManagedObjectID<Blog>!
+
+    override func setUp() async throws {
+        repository = PostRepository(coreDataStack: contextManager)
+
+        let loggedIn = try await signIn()
+        blogID = try await contextManager.performAndSave {
+            let blog = try BlogBuilder($0)
+                .with(dotComID: 42)
+                .withAccount(id: loggedIn)
+                .build()
+            return TaggedManagedObjectID(blog)
+        }
+    }
```

Comment:
I'm aware of this weird XCTest behaviour. But, I don't think we should avoid using instance variables just because this weird behaviour exists? Of course, making everything transient and deallocated after each test run would reduce chances of potential issues. But that feels like a lot afford, in terms of coding, especially if we know those instance variables—even though being "leaked"–should not introduce problems.

Reply:
> But that feels like a lot afford, in terms of coding

Is it extra effort, though?

Most of the coding has to be done regardless. One version has it in `make...` functions, the other in the `setup`. 

There's an additional advantage: _each test method is self contained and sets up all and only what it needs._

I've seen test (not sure if in this codebase, but I can look it up) where steps taken in the `setUp` had to be partially undone in individual tests because of different state expectations. Of course, one could argue that if a test has to do that much setup, maybe the problem is in the software design...

Anyways, I think this is a conversation broader than the scope of this PR 😄 Thank you for sharing your point of view. I'll P2 at some point.

Reply:
> Is it extra effort, though?

Maybe not too bad in this particular test case where there is only two properties. But for other test cases that have more properties, we'll have to repeat the instantiation statements in all test functions, which feels like a bit too much...

My point is this setting up test environment code belongs to the `setUp` function, and it'd be much nicer to keep them there. If we have to make a convention, an alternative to moving those instantiation code out of `setUp` and into the test functions is, setting those properties to nil in `tearDown`. We kind of get benefits from both world: setting up test environment code is consolidated in the `setUp` function and they won't be kept in memory forever due to how XCTest runs tests.